compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1677,6 +1679,40 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                     }
                     _ => {}
                 }
+            } else if let Some(attr) =
+                self.tcx().get_attr(def_id, sym::rustc_intrinsic_const_vector_arg)
+            {
+                match attr.meta_item_list() {
+                    Some(items) => {
+                        items.into_iter().for_each(|item: rustc_ast::NestedMetaItem| match item {
+                            rustc_ast::NestedMetaItem::Lit(rustc_ast::MetaItemLit {
+                                kind: rustc_ast::LitKind::Int(index, _),
+                                ..
+                            }) => {
+                                if index >= args.len() as u128 {
```

Comment:
to avoid checking the index separately from indexing, use

```suggestion
                                let Some(arg) = args.get(index.get().try_into().unwrap()) {
```

Reply:
I believe I have incorporated all the requested changes to this file in the following commit: 0309238ba4c1114d96d6a529ac461a7f20676381